lib/propagation/w3c.js

```diff
@@ -0,0 +1,92 @@
+/* global require, exports */
+const { TRACE_PARENT_HEADER, parseTraceParent } = require("@opentelemetry/core"),
+  util = require("./util");
+// requiring the OTEL trace header key from the api
+// as well as the parser
+exports.TRACE_HTTP_HEADER = TRACE_PARENT_HEADER;
+
+const VERSION = "00";
+exports.VERSION = VERSION;
+
+exports.unmarshalTraceContext = unmarshalTraceContext;
+
+function addToContext(base, headers) {
+  Object.assign(base, headers);
+}
+
+// unmarshalTraceContext takes an incoming header
+// and returns a context in a format the beeline can use
+
+// OTEL context structure         Beeline context
+// traceId                   -->  traceId
+// spanId                    -->  parentSpanId
+// traceFlags                -->  context.traceFlags (nested field in context: {})
+// isRemote (optional)       -->  context.isRemote (nested field in context: {})
+// traceState (optional)     -->  context.traceState (nested field in context: {})
+
+function unmarshalTraceContext(header) {
+  const regex = new RegExp(`^${VERSION}`, "g");
+
+  // check for supported version
+  if (!header.match(regex)) {
+    // bail if not supported
+    return;
+  }
+
+  return unmarshalTraceContextv1(header);
+}
+
+// v1 here refers to honeycomb, the context structure
+function unmarshalTraceContextv1(header) {
+  // pull out required fields, ... for all optional
+  // using spread operator defaults all values to undefined for invalid header values
+  const { traceId, spanId, traceFlags, ...optionalFields } = { ...parseTraceParent(header) };
```

Comment:
```suggestion
  const { traceId, spanId, traceFlags, ...optionalFields } = parseTraceParent(header);
```

if we're destructuring, we don't also need to clone the return value.

Reply:
The spread/clone here gives us the fallback to return `traceId: undefined` for instance instead of the return from parseTraceParent returning undefined (which will break the destructuring). Just thinking how else to work around this. `parseTraceParent` will return `null` if something is found invalid or missing. We could nix the destructure statement altogether, but the spread is really nice for pulling out those extra fields that may or may not be there.

Reply:
Oh we could test for a null return first, and then move on to this?
```
const parsed = parseTraceParent(header);
if (!parsed) {
  return;
}

const { traceId, spanId, traceFlags, ...optionalFields } = parsed;
```

Reply:
Yup this passes, I'll push up the change so you can check it out